### DISCUSSION TITLE
Using BaseModel class instead Schema

### DISCUSSION THREAD

**Discussion by HryhorenkoVitalii** [vitalik/django-ninja#985] [General]
Hello, I have a question about how safe is it to use BaseModel from pydentic instead Schema? 
What problems might I encounter now or in the future with this approach?

Thank you

**Comment by vitalik** [comment]
100% safe

The goals of Schema class are:

 - Naming convention - in django Model means db model - so introducing other "models" is a big naning clash
 - Schema is inherited from Base model but deeply integrated into django - it automatically serializes FileFields, Querysets, managers
 - it has somewhat better resolve_xxx method that takes context (and that context contain django request object)
 - it have backwards compatibility with pydantic v1 - so if you started with pydantic-v1  and used Schema - you should have easier time migrating to pydantic2 (and I think for future it will stay the same - we will try to invest better time into easier version migrations rather than major bumps)

**Comment by HryhorenkoVitalii** [comment]
Thank you for your answer )

**Comment by mathiasertl** [comment]
Also just wondered about this and stumbled upon this discussion. Could we metnion this in the documentation explicitly, so that it's "officially supported"? Otherwise it feels a bit like using something that just "accidentally works"...
